### PR TITLE
Add a introduction documents for using Intel Neural Compressor to conduct post-training quantization

### PR DESCRIPTION
**Pull request by XuhuiRen**
## What does this PR do?
This pr includes an introduction about how to use Intel Neural Compressor to accelerate the model inference speed on Intel hardware with post-training quantization

### Does your PR introduce any breaking changes? If yes, please list them.

Yes. Currently, PyTorch Lightning did not include the feature of post-training quantization function. This pr will introduce a new way that uses Intel Neural Compressor as the third-party api to quantize the PyTorch Lightning model.

### CODE REVIEW

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (ftian1) [inline docs/source-pytorch/advanced/post-training quantization.rst:7]: should be Intel(R) Neural Compressor

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (ftian1) [inline docs/source-pytorch/advanced/post-training quantization.rst:91]: pls check if "conduct quantization " is a well-known word for user/developer

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (ftian1) [inline docs/source-pytorch/advanced/post-training quantization.rst:114]: this is a little ugly. @PenghuiCheng  is it what you mentioned before? that's the fx tracing saves temp model in other places, we have no way to do in-place model modification?

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (kevinintel) [inline docs/source-pytorch/advanced/post-training quantization.rst:143]: you can add model size

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (kevinintel) [inline docs/source-pytorch/advanced/post-training quantization.rst:7]: please add link of INC

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (kevinintel) [inline docs/source-pytorch/advanced/post-training quantization.rst:87]: can we simplify it?

**Review by Borda — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]
LGTM

**Comment by hshen14** [comment]
Hi @tchaton, @lantiga @edenlightning @awaelchli . this PR is to follow up the previous comments from https://github.com/Lightning-AI/lightning/pull/10793. Please review. Thanks.

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post-training quantization.rst:1]: @XuhuiRen There seems to be a space in the file name. Could you remove it?

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post-training quantization.rst:2]: The page is currently not referenced anywhere. We need place it somewhere, otherwise it won't be found.

Perhaps it could be integrated here: https://pytorch-lightning.readthedocs.io/en/stable/advanced/pruning_quantization.html?highlight=quantization

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post-training quantization.rst:34]: Aren't these getting installed through

`pip install neural-compressor`
? 
We wouldn't want the user to install these manually.

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/post-training quantization.rst:1]: done

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/post-training quantization.rst:2]: Hi, Adrian, I just update this commit and add some descriptions in index.rst and pruning_quantization.rst. I still keep the main storyline of the original introduction part but append the description about Post-training quantization with Intel Neural Compressor. Hope for your kind suggestions on my changes.

File: docs/source-pytorch/advanced/pruning_quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/pruning_quantization.rst:125]: I think here in this paragraph here we could easily link to the main document `advanced/post_training_quantization`. What do you think?

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post_training_quantization.rst:58]:
```suggestion
PyTorch Lightning model

```

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/advanced/post-training quantization.rst
Comment (awaelchli) [reply docs/source-pytorch/advanced/post-training quantization.rst]: Thank you! I will confirm with @edenlightning and then we can merge it!

File: docs/source-pytorch/advanced/pruning_quantization.rst
Comment (hshen14) [reply docs/source-pytorch/advanced/pruning_quantization.rst]: Thanks @awaelchli. May I know how to add to the right link to "advanced/post_training_quantization" when the doc being compiled?

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/post_training_quantization.rst]: Done.

**Comment by hshen14** [comment]
Hi @williamFalcon , please help have a review. Thanks.

**Comment by Borda** [comment]
I think it is rather on @lantiga to have look :)

File: docs/source-pytorch/advanced/pruning_quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/pruning_quantization.rst]: Hi, Adrian, this link problem has been fixed in the update.

**Comment by hshen14** [comment]
Hi @Borda @awaelchli , we are seeing some test failures which seems related with doc issue. Could you please point out the details? We will take a look and work on the fixing. Thanks.

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post_training_quantization.rst:180]: Is it meant to say "Technical Support"?

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (awaelchli) [inline docs/source-pytorch/advanced/post_training_quantization.rst:51]: "Usage" and the following sections should probably be at the same level as "Installation"

```suggestion

*****
Usage
*****
```

**Comment by awaelchli** [comment]
> Hi @Borda @awaelchli , we are seeing some test failures which seems related with doc issue. Could you please point out the details? We will take a look and work on the fixing. Thanks.


These were the issues reported in the docs build:
```
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:63: ERROR: Unexpected indentation.
[449](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:450)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:69: ERROR: Unexpected indentation.
[450](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:451)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:75: WARNING: Inline strong start-string without end-string.
[451](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:452)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:74: WARNING: Inline strong start-string without end-string.
[452](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:453)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:79: WARNING: Definition list ends without a blank line; unexpected unindent.
[453](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:454)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:94: ERROR: Unexpected indentation.
[454](https://github.com/Lightning-AI/lightning/actions/runs/3769683235/jobs/6409055237#step:8:455)
/home/runner/work/lightning/lightning/docs/source-pytorch/advanced/post_training_quantization.rst:95: WARNING: Block quote ends without a blank line; unexpected unindent.
```

You can see my last commits how I resolved theses.

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/post_training_quantization.rst]: Thanks for your comments. The revision has been added to the update.

File: docs/source-pytorch/advanced/post_training_quantization.rst
Comment (XuhuiRen) [reply docs/source-pytorch/advanced/post_training_quantization.rst]: Done.

**Comment by awaelchli** [comment]
@XuhuiRen You force-pushed and deleted my commit that fixed several formatting issues. See my comment [here](https://github.com/Lightning-AI/lightning/pull/16085#issuecomment-1364899945).
I pushed it again. This fixes the misaligned text and broken code blocks.

**Comment by XuhuiRen** [comment]
@awaelchli  Really appreciate your help. It seems all errors were fixed. May I ask does it is available to be merged into the master?

**Comment by awaelchli** [comment]
@XuhuiRen Yes, I'd like to merge it soon. I'm waiting on approval from @edenlightning, because this PR is adding a new item in the navigation menu. 

Hopefully next week. Thanks for the patience, we won't forget it!

**Comment by XuhuiRen** [comment]
@awaelchli OK, I got it. Thanks a lot!

File: docs/source-pytorch/index.rst
Comment (edenlightning) [inline docs/source-pytorch/index.rst:254]: Since we already have a quantization page in the menu, this link is redundant.

**Review by edenlightning — CHANGES_REQUESTED** [review]
Thanks for your PR :) 
I would suggest removing the link from Glossary, other than that LGTM!

File: docs/source-pytorch/index.rst
Comment (hshen14) [reply docs/source-pytorch/index.rst:254]: @edenlightning The 'post_training_quantization' page is newly added there is a need to expose the link from up-level user document per previous comments. Any suggestions on how to make this page easily accessible by users?

**Pull request merged**
